src/crypto/equihash.h

```diff
@@ -89,17 +101,46 @@ class FullStepRow : public StepRow<WIDTH>
 
 public:
     FullStepRow(const unsigned char* hashIn, size_t hInLen,
-                size_t hLen, size_t cBitLen, eh_index i);
+                size_t hLen, size_t cBitLen, eh_index i) :
+            StepRow<WIDTH> {hashIn, hInLen, hLen, cBitLen} {
+        EhIndexToArray(i, hash+hLen);
+    }
     ~FullStepRow() { }
 
     FullStepRow(const FullStepRow<WIDTH>& a) : StepRow<WIDTH> {a} { }
+
     template<size_t W>
-    FullStepRow(const FullStepRow<W>& a, const FullStepRow<W>& b, size_t len, size_t lenIndices, int trim);
-    FullStepRow& operator=(const FullStepRow<WIDTH>& a);
+    FullStepRow(const FullStepRow<W>& a, const FullStepRow<W>& b, size_t len, size_t lenIndices, int trim) :
```

Comment:
What is all this stuff being changed that doesn't have to do with conditional compilation of the mining code?